### DISCUSSION TITLE
Modeling_options.yaml inputs not making their way to simulation?

### DISCUSSION THREAD

**Discussion by mlidGWT** [NLRWindSystems/WEIS#244] [General]
Hey other WEIS users. Has anyone else experienced defining an input in `modeling_options.yaml` (based on it being available in the [modeling_schema.yaml](https://github.com/WISDEM/WEIS/blob/main/weis/inputs/modeling_schema.yaml)), and yet not had that defined input make its way to the simulation? 

I have several examples of this from various 'layers' within the `modeling_options.yaml`, but most recently I was trying to run with a particular wind file to take a look in more detail than from a previous run. Going forward I'll skip using WEIS for this type of simulation and run directly in OpenFAST, but I still feel like WEIS should have behaved differently than it did. 

Within my DLC definition block, I added a line for `wind_file:` as provided on [line 3065](https://github.com/WISDEM/WEIS/blob/5332c31ebcc658f269c16a175656a8b35f4fe79c/weis/inputs/modeling_schema.yaml#L3065) of the schema, and pointed it to the full path of the desired .bts file. 

And yet, when WEIS wrote the simulation files, the case_matrix shows a completely different wind seed and wind filename. I have no idea why. It ran everything else how I had specified (wind speed, wind direction, etc.). Are there certain conflicts or prerequisites for some of the inputs in the modeling_schema?

**Comment by gbarter** [comment]
Thank you for trying to improve the WEIS codebase!  What you are chasing should definitely work, however we have all encountered the same issue where it is a silly mistake in the yaml syntax or the python syntax that references the nested options file.  There is also the possibility that the correct OpenFAST flag wasn't set to use the bts-file you provided.  For debugging, carefully check the OpenFAST input files and don't be afraid to set breakpoints in the code and use the python debugger to inspect variables directly.

**Reply by mlidGWT** [reply]
So in this case, I do think it was because of my OpenFAST input `InflowWind.dat` file (I'm using an openfast directory to run WEIS), but that raises a new question. I thought that WEIS would only use the specified OpenFAST .dat files as a starting point, and would overwrite inputs that are defined in the `modeling_options.yaml` - for instance, it did successfully overwrite the input `InflowWind.dat` file's `WindType` from 1 to 3 for the simulation in question, but it did not then overwrite `FileName_BTS`. Is that a bug? How am I supposed to know which inputs will and won't be overwritten by WEIS?

For more detail, it seems like WEIS *did* grab the right filepath as specified, and part of the filename, but the seed number is the same as from my input InflowWind file: 
- Input InflowWind.dat filename: (different path)/IEA15_NTM_U4.000000_Seed1501552846.0.bts
- modeling_options.yaml defined filename: (correct path)/DLC62_1EWM50_U70.000000_Seed249069577.0.bts
- WEIS-written InflowWind filename: (correct path)/DLC62_1EWM50_U70.000000_Seed1501552846.0.bts

Do I need to specify both the `wind_file` and `wind_seed`? (I'll try this now; If so, could the schema file parameter descriptions be updated to reflect this dependency?)

**Reply by mlidGWT** [reply]
The wind seed seemed to take when I defined both the `wind_file` and `wind_seed` (it may have been sufficient to only specify `wind_seed`), though as I mentioned in Issue #245 it didn't seem to work for the HydroDyn wave seeds.
